### PR TITLE
Added ObjectList::fromArray.

### PR DESCRIPTION
**Pull request by pies**

### CODE REVIEW

File: src/ObjectList.php [hunk 1]
```diff
@@ -362,6 +363,60 @@ private function permute(&$out, $list, $perms = [], $max = null)
...
+    /**
+     * Create a single member element from serialized data.
+     *
+     * @param mixed $data
+     * @return mixed
+     * @throws Exception
+     */
+    public static function deserializeItem($data)
+    {
+        throw new Exception(
```
Comment (OneWebPro) [inline src/ObjectList.php]: Stupid php. We can't have `public static abstract` right?

File: src/ObjectList.php [hunk 2]
```diff
@@ -362,6 +363,60 @@ private function permute(&$out, $list, $perms = [], $max = null)
...
+    /**
+     * Turn this object into an array using toArray method on each element if they have it.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $out = [];
+        foreach ($this->getArrayCopy() as $key => $value) {
+            if (is_object($value) && method_exists($value, 'toArray')) {
```
Comment (OneWebPro) [inline src/ObjectList.php]: I think we can optimize it a little and not check on each element if `method_exists`.  Maybe we can check it only once? But maybe we shouldn't think about micro optimization.

**Comment by OneWebPro** [comment]
Looks ok, only 1 comment - more to think of then to change. :+1:

File: src/ObjectList.php (hunk 1 above)
Comment (pies) [inline src/ObjectList.php]: Well, it's not a mandatory feature of ObjectList :)

File: src/ObjectList.php (hunk 2 above)
Comment (pies) [inline src/ObjectList.php]: Well, it's possible for objects not to be of the same class (you can validate input by interface, or not at all, if you so choose).

**Pull request merged**